### PR TITLE
[WIP] automatic datagen

### PR DESCRIPTION
**Pull request by xf8b**
this pr:
- fixes CI not working
- makes datagen automatic
- ***changes the registry name of an item***

this pr **DOES NOT**:
- add any new datagen
- fix any bugs with the previous datagen

**xf8b makes no guarantees that this PR is bug free**

***THIS PR WILL REMOVE ALL INSTANCES OF NECROMINION PAGES OR BUNDLES IN YOUR WORLDS***

### CODE REVIEW

**Comment by xf8b** [comment]
how do i mark as wip

**Comment by xf8b** [comment]
there we go, why does github have to fucking put it in a super small place

**Comment by xf8b** [comment]
right, forgot to mention:
***THIS PR DOES NOT ADD ANY NEW DATAGEN***
if you wish for me to createe another pr, ***please tell me***.
my code for datagen is at https://github.com/xf8b/MoreFeatures/tree/1.15.2/src/main/java/io/github/xf8b/morefeatures/datagen

**Review by TheOnlyTails — APPROVED** [review]
This is great. Also, you can't automate recipe and loot table generation. They're unique for each item and/or block.

**Comment by xf8b** [comment]
irrt

**Comment by xf8b** [comment]
bruh why did you unmark as draft

no

**Comment by TheOnlyTails** [comment]
my mistake
sorry

**Comment by xf8b** [comment]
also 

> can't automate loot table generation

wat
https://github.com/DenDragon23/LovecraftPlus/pull/6/files#diff-98b15879a6a1aeff2d80a43dc7659ea0a878075919299f023876a4ec13ac3686

**Comment by TheOnlyTails** [comment]
idk
I was tired

**Comment by xf8b** [comment]
go to sleep or irritate

**Comment by TheOnlyTails** [comment]
no
I was tired from sleeping
it's 17:40 here rn

**Comment by TheOnlyTails** [comment]
anyway, can you add the blockstates/models generation?

**Comment by xf8b** [comment]
ok

**Comment by xf8b** [comment]
give me 1 day i got other stuff

**Comment by TheOnlyTails** [comment]
ok

File: src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java [hunk 1]
```diff
@@ -19,6 +20,16 @@ public RecipesGenerator(DataGenerator generator) {
...
     @Override
     public void registerRecipes(Consumer<IFinishedRecipe> consumer) {
+        // TODO: make this and structure loot tables automatic, if possible
+
+        BlockRegistries.BLOCKS
+                .getEntries()
+                .stream()
+                .map(RegistryObject::get)
+                .forEach(block -> {
+                    // TODO
```
Comment (TheOnlyTails) [inline src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java:30]: Here's what I think you should do - make a map, and for each item, make a "base" item. that way you can generate stuff for slabs, stairs, bricks, and more. then we could just add edge cases for special items. also, do the same for tools.

File: src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java [hunk 2]
```diff
@@ -19,6 +20,16 @@ public RecipesGenerator(DataGenerator generator) {
      */
     @Override
     public void registerRecipes(Consumer<IFinishedRecipe> consumer) {
+        // TODO: make this and structure loot tables automatic, if possible
+
+        BlockRegistries.BLOCKS
```
Comment (TheOnlyTails) [inline src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java:25]: `BlockRegistries`? Don't you mean `ItemRegistries`?

File: src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java (hunk 1 above)
Comment (xf8b) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java:30]: i dont remember commiting this

File: src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java (hunk 2 above)
Comment (xf8b) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java:25]: idk i forgot

File: src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java (hunk 1 above)
Comment (xf8b) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java:30]: ohhhh fuck i commited it while updating mappings

Comment (TheOnlyTails) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java:30]: well
you did

File: src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java (hunk 2 above)
Comment (xf8b) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java]: yes i did mean that, look at the recipes below

Comment (xf8b) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java]: remember: i yeeted manual blockitem registration

File: src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java (hunk 1 above)
Comment (xf8b) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java]: a "base item"? what? i can just do `instanceof SlabBlock` or something

i mean, unless i needa get like the planks of a slab block then yes i will create an interface for that

Comment (xf8b) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java]: also itembase bad :irritated:

Comment (TheOnlyTails) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java]: yes
making an interface will be good

File: src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java (hunk 2 above)
Comment (TheOnlyTails) [reply src/main/java/com/github/hauntedchest/lovecraftplus/datagen/RecipesGenerator.java]: yes
I see what you mean. that's good!

**Comment by xf8b** [comment]
comment

smh

**Comment by xf8b** [comment]
an idiot

smh

**Comment by xf8b** [comment]
closing because of resignation

**Pull request closed without merging**